### ISSUE TITLE
declare some third party variable like 'moment' in typing.d.ts, and run build library also throw: Cannot find name 'moment'

### ISSUE THREAD

**Issue by hpyou** [ng-packagr/ng-packagr#1326]
## Type of Issue

```
[ x] Bug Report
[ ] Feature Request
```

## Description

declare some third party variable like 'moment' in typing.d.ts, and run build library also throw: Cannot find name 'moment'

#### Expected Behaviour

how can i config this third party variable to build libary

#### Version Information

```
$ node_modules/.bin/ng-packagr --version
ng-packagr:            5.3.0
@angular/compiler:     8.0.2
rollup:                1.16.2
tsickle:               0.35.0
typescript:            3.4.5

```

**Comment by raphael22** [comment]
Welcome to my world of 1+ year of "nobody care".
https://github.com/ng-packagr/ng-packagr/issues/744

**Comment by alan-agius4** [comment]
Hi there?

How are you using ng-packagr? Is it with a custom tsconfig, with the CLI or with the internal tsconfig?

Can you share a reproduction? 

Thanks.

**Comment by raphael22** [comment]
I think @hpyou and I trying to achieve a final umd bundle which can be used as it.
Meaning we need a way to declare embeded dependencies.
We don't want to execute a npm install after an ng build, so "dependencies" or "bundleDependencies" won't solve this.
Only way to achieve this now is to going through a custom rollup/browserify pass.
TLDR: How to bundle angular library to be used on the fly on a deployed angular application ?

**Comment by alan-agius4** [comment]
@raphael22, Personally I would advice against this pattern for a couple of reasons

- When you bundle moment, and another library or the app itself depends on moment, the user will end up with two copies of moment in their application.
- You give zero flexibility to which version of a dependent library the users installs.
- All of a sudden if moment has a new bug fix, it will also be your responsibility to update and release your library

Can you please elaborate why you don't want to do `npm install`?

**Comment by raphael22** [comment]
Hi, thanks for quick response.

It's a matter of taking both benefit of peerDependencies and dependencies in a production context.

_As an Angular deployed applications hub, I want to be able to import plugins angular applications._
_I don't want them to copies peerDependencies, but I don't want to install there own dependencies._

I want to be able to develop a new plugin application without having to reference it's dependencies in my hub application and without to redeploy.

A nice final umd file which can be consume by another ALREADY built angular application.
I can already achieve this with extra steps by installing dependencies in dist folder and using browserify on top on that but I guess it's could be bone by some rollup step on build.
